python_modules/dagster-pipes/dagster_pipes_tests/test_context.py

```diff
@@ -249,3 +252,19 @@ def test_multiple_close():
     # `close` is idempotent, multiple calls should not raise an error
     context.close()
     context.close()
+
+
+def test_message_json_schema_validation():
+    message = {
+        PIPES_PROTOCOL_VERSION_FIELD: PIPES_PROTOCOL_VERSION,
+        "method": "foo",
+        "params": {"bar": "baz"},
+    }
+    jsonschema.validate(message, get_pipes_json_schema("message"))
+
+
+def test_json_schema_rejects_invalid():
+    with pytest.raises(jsonschema.ValidationError):
+        jsonschema.validate({"foo": "bar"}, get_pipes_json_schema("context"))
+    with pytest.raises(jsonschema.ValidationError):
+        jsonschema.validate({"foo": "bar"}, get_pipes_json_schema("message"))
```

Comment:
We need more tests here. We should be building a payload using the APIs used by the framework and validating them against the json schema.

Reply:
I added a test to explicitly validate the output of `build_external_execution_context_data`. This is also already tested a bunch of times above in any test that uses the helper `_make_external_execution_context`, which creates various flavors of `PipesContextData` (e.g. with and without partition key).

In the external -> orchestration direction, the existing protocol only defines the high level `PipesMessage` rather than all the individual messages, and that is tested in `test_message_json_schema_validation`.

```
class PipesMessage(TypedDict):
    """A message sent from the external process to the orchestration process."""

    __dagster_pipes_version: str
    method: str
    params: Optional[Mapping[str, Any]]
```